libdeno/BUILD.gn

```diff
@@ -0,0 +1,83 @@
+# Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import("./deno.gni")
+import("//third_party/v8/gni/v8.gni")
+
+# Only functionality needed for libdeno_test and snapshot_creator
+# In particular no flatbuffers, no assets, no rust, no msg handlers.
+# Because snapshots are slow, it's important that snapshot_creator's
+# dependencies are minimal.
+static_library("libdeno") {
+  sources = [
+    "api.cc",
+    "binding.cc",
+    "deno.h",
+    "file_util.cc",
+    "file_util.h",
+    "internal.h",
+  ]
+  public_deps = [
+    "//third_party/v8:v8_monolith",
+  ]
+  configs += [ ":deno_config" ]
+}
+
+executable("snapshot_creator") {
+  sources = [
+    "snapshot_creator.cc",
+  ]
+  deps = [
+    ":libdeno",
+  ]
+  configs += [ ":deno_config" ]
+}
+
+v8_executable("test_cc") {
+  testonly = true
+  sources = [
+    "file_util_test.cc",
+    "libdeno_test.cc",
+    "test.cc",
+  ]
+  deps = [
+    ":create_snapshot_libdeno_test",
+    ":libdeno",
+    "//testing/gtest:gtest",
+  ]
+  data = [
+    "$target_gen_dir/snapshot_libdeno_test.bin",
+  ]
+  snapshot_path = rebase_path(data[0], root_build_dir)
+  defines = [ "SNAPSHOT_PATH=\"$snapshot_path\"" ]
+  configs = [ ":deno_config" ]
+}
+
+# Generates $target_gen_dir/snapshot_libdeno_test.bin
+create_snapshot("libdeno_test") {
+  testonly = true
+  js = "libdeno_test.js"
+}
+
+config("deno_config") {
+  include_dirs = [ "//third_party/v8" ]  # This allows us to v8/src/base/ libraries.
+  configs = [ "//third_party/v8:external_config" ]
+  if (is_debug) {
+    defines = [ "DEBUG" ]
+  }
+
+  # Targets built with the `rust_executable()` template automatically pick up
+  # these dependencies, but those built with `executable()` need them when they
+  # have Rust inputs. Currently, there's only one such target, `test_cc`.
+  if (is_mac) {
+    libs = [ "resolv" ]
```

Comment:
This config directive is unnecessary for libdeno, since it doesn't do any DNS resolution.
Rather they are required for our Rust code to work. I would suggest to move them to a separate config in //BUILD.gn.

Reply:
Removed completely since we no longer have any `executable()` targets that depend on rust code.